log/sequencer.go

```diff
@@ -166,8 +171,12 @@ func (s Sequencer) createRootSignature(ctx context.Context, root trillian.Signed
 // which will fail if the tx was committed. Should only do this if we can hide the details of
 // the underlying storage transactions and it doesn't create other problems.
 func (s Sequencer) SequenceBatch(ctx context.Context, limit int) (int, error) {
-	tx, err := s.logStorage.Begin(ctx)
+	logID, err := util.LogID(ctx)
```

Comment:
Yeah, `util.LogIDPrefix(ctx)` was only intended for diagnostics (and continues to work if handed a context without a value), so extending the idea to require the value to be present is probably brittle.

(Aside: I'm sure I found some general Go advice along these lines, but I can't find the link now.)

BTW, I don't think "the expected context.Background()" is accurate -- there should definitely never be an expectation that `ctx == context.Background()`.